src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copierStudents = [...students];
+  const sortOrder = order === 'asc' ? 1 : -1;
+
+  type Callback = (prev: number, cur: number) => number;
+
+  const sumCallback: Callback = (prev, cur) => prev + cur;
+
+  switch (sortBy) {
+    case SortType.Name:
+      copierStudents.sort(
+        (a: Student, b: Student) => a.name.localeCompare(b.name) * sortOrder,
+      );
+      break;
+
+    case SortType.Surname:
+      copierStudents.sort(
+        (a: Student, b: Student) => a.surname.localeCompare(b.surname)
+        * sortOrder,
+      );
+      break;
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      copierStudents.sort(
        (a: Student, b: Student) => a[sortBy].localeCompare(b[sortBy])
        * sortOrder,
      );
      break;
```
checklist p.3 [CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.